develop/core/views.py

```diff
@@ -21,35 +20,33 @@ def get_queryset(self):
         return self.model.on_site.all()
 
 
-class ProductAccessView(ProductRequiredMixin, TemplateView):
+class ProductAccessView(SetSiteToRequestMixin, ProductRequiredMixin, TemplateView):
     model = Offer
     template_name = "core/product_use.html"
 
     def get(self, request, *args, **kwargs):
         context = super().get_context_data(**kwargs)
-        site = get_current_site(request)
-        context['object'] = self.model.objects.get(site=site, slug=kwargs['slug'])
+
+        context['object'] = self.model.objects.get(site=request.site, slug=kwargs['slug'])
         return render(request, self.template_name, context)
 
     def get_product_queryset(self):
         """
         Method to get the Product(s) needed for the check.  Can be overridden to handle complex queries.
         """
-        site = get_current_site(request)
-        return self.model.objects.filter(site=site, slug=self.kwargs['slug']).get().products.all()
+        return self.model.objects.filter(site=self.request.site, slug=self.kwargs['slug']).get().products.all()
 
-class AccountView(LoginRequiredMixin, TemplateView):
+class AccountView(LoginRequiredMixin, SetSiteToRequestMixin, TemplateView):
     template_name = "core/account.html"
 
     def get(self, request, *args, **kwargs):
         context = super().get_context_data(**kwargs)
-        site = Site.objects.get_current()
         
-        customer_profile, created = request.user.customer_profile.get_or_create(site=site)
+        customer_profile, created = request.user.customer_profile.get_or_create(site=request.site)
         subscriptions = customer_profile.get_recurring_receipts()
 
         context['payments'] = customer_profile.payments.filter(success=True)
-        context["offers"] = Offer.objects.filter(site=site, available=True).order_by('terms')
+        context["offers"] = Offer.objects.filter(site=request.site, available=True).order_by('terms')
```

Comment:
use the new function here to get the site object instead of modifying the request object.